cmd/gradleExecuteBuild.go

```diff
@@ -189,6 +191,25 @@ func runGradleExecuteBuild(config *gradleExecuteBuildOptions, telemetryData *tel
 		return err
 	}
 
+	log.Entry().Debugf("creating build settings information...")
+
+	dockerImage, err := GetDockerImageValue(stepNameForBuildSettings)
+	if err != nil {
+		return err
+	}
+
+	gradleConfig := buildsettings.BuildOptions{
+		CreateBOM:         config.CreateBOM,
+		Publish:           config.Publish,
+		BuildSettingsInfo: config.BuildSettingsInfo,
+		DockerImage:       dockerImage,
+	}
+	buildSettingsInfo, err := buildsettings.CreateBuildSettingsInfo(&gradleConfig, stepNameForBuildSettings)
+	if err != nil {
+		log.Entry().Warnf("failed to create build settings info: %v", err)
```

Comment:
Why do we fail above, but not here?

Reply:
the buildSettings info gathering can fail, but that must fail the step, its a compliance information fetch fail, but i am not sure if the entire step should fail for that reason ?

Reply:
It works the same way across all build steps. 
If we have a warning instead of err, in case the method above fails, dockerImage in build-settings will be an empty string, but in fact gradle image is used.
It is compliance information, not sure if having wrong docker image info is ok.